test/mockupdb/test_handshake.py

```diff
@@ -12,14 +12,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from mockupdb import (MockupDB, OpReply, OpMsg, OpMsgReply, OpQuery, absent,
+                      Command, go)
 
-from mockupdb import MockupDB, OpReply, OpMsg, absent, Command, go
 from pymongo import MongoClient, version as pymongo_version
 from pymongo.errors import OperationFailure
+from pymongo.server_api import ServerApi, ServerApiVersion
+from bson.objectid import ObjectId
 
 import unittest
 
 
+def test_hello_with_option(self, protocol, **kwargs):
+    hello = "ismaster" if isinstance(protocol(), OpQuery) else "hello"
+    # `db.command("hello"|"ismaster")` commands are the same for primaries and
+    # secondaries, so we only need one server.
+    primary = MockupDB()
+    # Set up a custom handler to save the first request from the driver.
+    self.handshake_req = None
+    def respond(r):
+        # Only save the very first request from the driver.
+        if self.handshake_req == None:
+            self.handshake_req = r
+        load_balanced_kwargs = {"serviceId": ObjectId()} if kwargs.get(
+            "loadBalanced") else {}
+        return r.reply(OpMsgReply(minWireVersion=0, maxWireVersion=13,
+                                  **kwargs, **load_balanced_kwargs))
+    primary.autoresponds(respond)
+    primary.run()
+    self.addCleanup(primary.stop)
+
+    # We need a special dict because MongoClient uses "server_api" and all
+    # of the commands use "apiVersion".
+    k_map = {("apiVersion", "1"):("server_api", ServerApi(
+                                        ServerApiVersion.V1))}
+    client = MongoClient("mongodb://"+primary.address_string,
+                         appname='my app', # For _check_handshake_data()
+                         **dict([k_map.get((k, v), (k, v)) for k, v
+                                 in kwargs.items()]))
+    
+    self.addCleanup(client.close)
+
+    # We have an autoresponder luckily, so no need for `go()`.
+    assert client.db.command(hello)["ok"], f"%s command failed" % (hello)
```

Comment:
FYI db.command() already checks the "ok" field and raises an error if the command failed so this line can be:
```
assert client.db.command(hello)
```

Reply:
Ok cool. Fixed.